wrapper/src/main/java/software/amazon/jdbc/plugin/limitless/LimitlessConnectionPlugin.java

```diff
@@ -238,8 +240,16 @@ private void initLimitlessRouterMonitorService() {
     }
   }
 
-  private Connection retryConnectWithLeastLoadedRouters(final List<HostSpec> limitlessRouters, final Properties props,
-      final Connection conn, final HostSpec hostSpec) throws SQLException {
+  private Connection retryConnectWithLeastLoadedRouters(
+      final List<HostSpec> limitlessRouters,
+      final Properties props,
+      final Connection inputConn,
+      final HostSpec hostSpec,
+      final JdbcCallable<Connection, SQLException> connectFunc) throws SQLException {
+    Connection conn = inputConn;
+    if (conn == null || conn.isClosed()) {
```

Comment:
That'll try to establish a connection to the same node as the driver has just tried and failed. How this call be different?

Reply:
Removed this and added more checks to throw exceptions when trying to use the connections.